### PR TITLE
ISSD-1497 [Marketplace] Creation of the table of licenses not yet created

### PR DESCRIPTION
**Pull request by eliasantosf**

### CODE REVIEW

**Comment by eliasantosf** [comment]
ci:test:sf

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/DashboardTable/PurchasedAppsDashboardTableRow.tsx [hunk 1]
```diff
@@ -166,6 +169,13 @@ export function PurchasedAppsDashboardTableRow({
 								</DropDown.Item>
 							</ClayTooltipProvider>
 						)}
+						<DropDown.Item
+							onClick={() =>
+								navigate(`/app/${productId}/licenses`)
+							}
+						>
+							Manage License Key(s)
```
Comment (silasbsi) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/DashboardTable/PurchasedAppsDashboardTableRow.tsx:177]: Please, use the i18n here.

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedAppsDashboard/Apps/AppOutlet.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+	return (
+		<div>
+			<ClayButton
+				className="mb-2"
+				displayType="unstyled"
+				onClick={() => navigate('/')}
+			>
+				<ClayIcon className="mr-2" symbol="order-arrow-left" />
+				Back to My Apps
```
Comment (silasbsi) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedAppsDashboard/Apps/AppOutlet.tsx:29]: Please, use i18n here.

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedAppsDashboard/Apps/Licenses.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import {DashboardEmptyTable} from '../../../components/DashboardTable/DashboardEmptyTable';
+
+const Licenses = () => {
+	return (
+		<DashboardEmptyTable
+			button={true}
+			buttonName={'Create License Key'}
+			description1={'Create new licenses and they will show up here.'}
+			icon={solutionsIcon}
+			title={'No Licenses yet'}
```
Comment (silasbsi) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedAppsDashboard/Apps/Licenses.tsx]: Please, use i18n here.

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedAppsDashboard/Apps/Licenses.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import solutionsIcon from '../../../assets/icons/bookmarks_icon.svg';
+
+import {DashboardEmptyTable} from '../../../components/DashboardTable/DashboardEmptyTable';
+
+const Licenses = () => {
+	return (
+		<DashboardEmptyTable
+			button={true}
+			buttonName={'Create License Key'}
+			description1={'Create new licenses and they will show up here.'}
```
Comment (silasbsi) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedAppsDashboard/Apps/Licenses.tsx]: Please, use i18n here.

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedAppsDashboard/Apps/Licenses.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+import solutionsIcon from '../../../assets/icons/bookmarks_icon.svg';
+
+import {DashboardEmptyTable} from '../../../components/DashboardTable/DashboardEmptyTable';
+
+const Licenses = () => {
+	return (
+		<DashboardEmptyTable
+			button={true}
+			buttonName={'Create License Key'}
```
Comment (silasbsi) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedAppsDashboard/Apps/Licenses.tsx]: Please, use i18n here.

**Comment by eliasantosf** [comment]
ci:test:sf

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedAppsDashboard/Apps/AppOutlet.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+			</ClayButton>
+			<ClayNavigationBar className="mb-4" triggerLabel={active}>
+				<ClayNavigationBar.Item active={active === AppTabType.DETAILS}>
+					<ClayButton
+						onClick={() => {
+							navigate(`/app/${productId}`);
+							setActive(AppTabType.DETAILS);
+						}}
+					>
+						Details
```
Comment (silasbsi) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedAppsDashboard/Apps/AppOutlet.tsx:39]: Please, use i18n here.

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedAppsDashboard/Apps/AppOutlet.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+					</ClayButton>
+				</ClayNavigationBar.Item>
+				<ClayNavigationBar.Item active={active === AppTabType.LICENSES}>
+					<ClayButton
+						onClick={() => {
+							navigate(`licenses`);
+							setActive(AppTabType.LICENSES);
+						}}
+					>
+						Licenses
```
Comment (silasbsi) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedAppsDashboard/Apps/AppOutlet.tsx:49]: Please, use i18n here.

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedAppsDashboard/Apps/Licenses.tsx [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+import {DashboardEmptyTable} from '../../../components/DashboardTable/DashboardEmptyTable';
+
+const Licenses = () => {
+	return (
+		<DashboardEmptyTable
+			button={true}
+			buttonName="Create License Key"
+			description1="Create new licenses and they will show up here."
+			icon={solutionsIcon}
+			title="No Licenses yet"
```
Comment (silasbsi) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedAppsDashboard/Apps/Licenses.tsx:16]: Please, use i18n here.

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedAppsDashboard/Apps/Licenses.tsx [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+import solutionsIcon from '../../../assets/icons/bookmarks_icon.svg';
+import {DashboardEmptyTable} from '../../../components/DashboardTable/DashboardEmptyTable';
+
+const Licenses = () => {
+	return (
+		<DashboardEmptyTable
+			button={true}
+			buttonName="Create License Key"
+			description1="Create new licenses and they will show up here."
```
Comment (silasbsi) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedAppsDashboard/Apps/Licenses.tsx:14]: Please, use i18n here.

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedAppsDashboard/Apps/Licenses.tsx [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+ */
+
+import solutionsIcon from '../../../assets/icons/bookmarks_icon.svg';
+import {DashboardEmptyTable} from '../../../components/DashboardTable/DashboardEmptyTable';
+
+const Licenses = () => {
+	return (
+		<DashboardEmptyTable
+			button={true}
+			buttonName="Create License Key"
```
Comment (silasbsi) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedAppsDashboard/Apps/Licenses.tsx:13]: Please, use i18n here.

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedAppsDashboard/Apps/enums/AppTabType.ts [hunk 11]
```diff
@@ -0,0 +1,9 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+export enum AppTabType {
```
Comment (silasbsi) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PurchasedAppsDashboard/Apps/enums/AppTabType.ts:6]: Please change this name to something like AppTabEnum.

**Review by silasbsi — CHANGES_REQUESTED** [review]
I added some suggestions

**Review by joanagrego — APPROVED** [review]

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by kevenleone** [comment]
Just starting reviewing :)

**Comment by kevenleone** [comment]
resent: https://github.com/brianchandotcom/liferay-portal/pull/143179

**Pull request closed without merging**